### PR TITLE
Support for PersianCalendar

### PR DESCRIPTION
**Pull request by spiegelp**
basic date picker support for `PersianCalendar`

solves issue #1077

### CODE REVIEW

File: MaterialDesignThemes.Wpf/MaterialDateDisplay.cs [hunk 1]
```diff
@@ -14,11 +14,19 @@ static MaterialDateDisplay()
...
+            UpdateComponents();
         }
 
         public static readonly DependencyProperty DisplayDateProperty = DependencyProperty.Register(
-            nameof(DisplayDate), typeof (DateTime), typeof (MaterialDateDisplay), new PropertyMetadata(default(DateTime), DisplayDatePropertyChangedCallback));
+            nameof(DisplayDate),
+            typeof (DateTime),
+            typeof (MaterialDateDisplay),
+            new PropertyMetadata(
+                System.Threading.Thread.CurrentThread.CurrentUICulture.Calendar != null ? System.Threading.Thread.CurrentThread.CurrentUICulture.Calendar.MinSupportedDateTime : default(DateTime),
```
Comment (Keboo) [inline MaterialDesignThemes.Wpf/MaterialDateDisplay.cs]: I don't think we want to have different default value based on culture. Rather I think we should take this culture into account at the places where the `DisplayDate` property is used.

Comment (Keboo) [reply MaterialDesignThemes.Wpf/MaterialDateDisplay.cs]: Yea I think that would be better. Just controlling it with a default value, also doesn't guard against someone setting a `DateTime` that is outside of the range. 
I think in these cases I am happy simply adjusting the invalid date to be in the valid range. I would also favor using the Language setting from the control over the culture specified on the thread. Something like:
```C#
MaterialDateDisplay materialDate = ...;
CultureInfo culture = materialDate.Language.GetSpecificCulture();
var minDateTime = culture.Calendar.MinSupportedDateTime;
var maxDateTime = culture.Calendar.MaxSupportedDateTime;

DateTime value = (DateTime) basevalue;
if (value < minDateTime)
{
    return minDateTime;
}

if (value > maxDateTime)
{
    return maxDateTime;
}

return value;
```

Comment (spiegelp) [reply MaterialDesignThemes.Wpf/MaterialDateDisplay.cs]: I implemented your suggestions.

**Comment by Cyrusbm** [comment]
I think this problem has not been resolved completely. Months names is not shown in Persian calendar type.
Untitled
Untitled2

**Comment by jamalhosseini** [comment]
This #1077 do not solved yet. the months name changed on header but not in calendar body. also days formation should be based on shamsi-date , but this is not.

@Cyrusbm have noticed this issue well.

**Comment by Keboo** [comment]
@jamalhosseini can you log a new issue for this, so that it doesn't get lost,

**Comment by jamalhosseini** [comment]
> @jamalhosseini can you log a new issue for this, so that it doesn't get lost,

you can see #1646

**Comment by jamalhosseini** [comment]
> @jamalhosseini can you log a new issue for this, so that it doesn't get lost,

@Keboo please see #1646

**Pull request merged**